src/main/java/co/cask/hydrator/plugin/batchSource/KafkaBatchSource.java

```diff
@@ -304,7 +304,10 @@ public Schema getMessageSchema() {
     }
 
     public void validate() {
-      getBrokerMap();
+      // brokers can be null since it is macro enabled.
+      if (kafkaBrokers != null) {
+        getBrokerMap();
```

Comment:
Yes. This method is called in PrepareRun() method. And it will fail with the info about which broker info is invalid in during initialize.